packages/tre/src/plugins/r2/validator.ts

```diff
@@ -67,6 +70,14 @@ export type R2Hashes = Record<
   Buffer | undefined
 >;
 
+function serialisedLength(x: string) {
+  //  Adapted from internal R2 gateway implementation
+  for (let i = 0; i < x.length; i++) {
+    if (x.charCodeAt(i) >= 256) return x.length * 2;
```

Comment:
What is this doing? This feels ...fishy. I may be wrong, but this _seems_ to be saying "if _any_ character has a character code greater than 256, return the string length doubled". If this is for checking the serialised length, what about serialising the string with a `TextEncoder` and checking the byte length?